### ISSUE TITLE
ulimit -- limitation on number of files open

### ISSUE THREAD

**Issue by stvhay** [linuxserver/docker-transmission#95]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

If you are new to Docker or this application our issue tracker is **ONLY** used for reporting bugs or requesting features. Please use [our discord server](https://discord.gg/YWrKVTn) for general support.

------------------------------

## Expected Behavior
I should be able to accept a lot of seeds and downloads without hitting /etc/limits.conf limits.

## Current Behavior
Participating in a large torrent, like archlinux ISO, causes system to essentially shutdown due to resource starvation

## Steps to Reproduce
1. Download an archlinux ISO.

## Environment
**OS:** Linux      
**CPU architecture:** x86_64      
**How docker service was installed:**      
Synology

## Command used to create docker container (run/create/compose/screenshot)
<img width="427" alt="image" src="https://user-images.githubusercontent.com/1269071/77216699-b0428200-6af2-11ea-9230-e1a269f5c072.png">

## Docker logs
[2020-03-21 01:19:18.624] Couldn't create socket: No file descriptors available (fdlimit.c:523)
[2020-03-21 01:19:18.624] Couldn't create socket: No file descriptors available (fdlimit.c:523)
[2020-03-21 01:19:18.624] Couldn't create socket: No file descriptors available (fdlimit.c:523)
[2020-03-21 01:19:18.625] Couldn't create socket: No file descriptors available (fdlimit.c:523)
[2020-03-21 01:19:18.625] Couldn't create socket: No file descriptors available (fdlimit.c:523)
[2020-03-21 01:19:19.625] Couldn't create socket: No file descriptors available (fdlimit.c:523)
[2020-03-21 01:19:19.625] Couldn't create socket: No file descriptors available (fdlimit.c:523)
[2020-03-21 01:19:19.625] Couldn't read "/transmission-web-control//tr-web-control/style/images/checking.png": No file descriptors available (utils.c:258)
[2020-03-21 01:19:19.625] Couldn't create socket: No file descriptors available (fdlimit.c:523)

**Comment by stvhay** [comment]
This limit seems to only apply to the proc running the actual daemon.

```
PID   USER     TIME  COMMAND
    1 root      0:00 s6-svscan -t0 /var/run/s6/services
   34 root      0:00 s6-supervise s6-fdholderd
  251 root      0:00 s6-supervise cron
  252 root      0:00 s6-supervise transmission
  257 root      0:00 /usr/sbin/crond -f -S -l 0 -c /etc/crontabs
 3137 root      0:00 /bin/bash
 4304 root      0:00 bash ./run
 4306 abc       9:53 /usr/bin/transmission-daemon -g /config -c /watch -f
 5074 root      0:00 ps

root@linuxserver-transmission1:/etc/init.d# cat /proc/4306/limits 
Limit                     Soft Limit           Hard Limit           Units     
Max cpu time              unlimited            unlimited            seconds   
Max file size             unlimited            unlimited            bytes     
Max data size             unlimited            unlimited            bytes     
Max stack size            8388608              unlimited            bytes     
Max core file size        unlimited            unlimited            bytes     
Max resident set          unlimited            unlimited            bytes     
Max processes             31015                31015                processes 
Max open files            1024                 1048576              files     
Max locked memory         65536                65536                bytes     
Max address space         unlimited            unlimited            bytes     
Max file locks            unlimited            unlimited            locks     
Max pending signals       31015                31015                signals   
Max msgqueue size         819200               819200               bytes     
Max nice priority         0                    0                    
Max realtime priority     0                    0                    
Max realtime timeout      unlimited            unlimited            us       
```

**Comment by stvhay** [comment]
Given this is something the user may want to control--I think the solution to this is to have an environment variable that can be set that the container is aware of.  I'm not entirely sure if this limit is being set by docker itself or something in the container.

**Comment by drizuid** [comment]
I'm unable to reproduce this on ubuntu or debian. I'll leave this open for anyone else willing to test with arch.

**Comment by stvhay** [comment]
Anything you want me to grab from the affected machine?

**Comment by drizuid** [comment]
We have a teammember who plans to test with syno and I imagine you'll have to wait on them since this appears to only affect syno (and potentially other nas units like syno)

**Comment by stevenboyd78** [comment]
It looks like I'm hitting this as well on a Qnap NAS.
`
[2020-04-04 11:48:24.139] Couldn't open "/library/Torrents/incomplete/After.the.Dark.2013.1080p.BluRay.x264.DTS-FGT/After.the.Dark.2013.1080p.BluRay.x264.DTS-FGT.mkv.part": No file descriptors available (fdlimit.c:191)
[2020-04-04 11:48:24.139] After.the.Dark.2013.1080p.BluRay.x264.DTS-FGT tr_fdFileCheckout failed for "/library/Torrents/incomplete/After.the.Dark.2013.1080p.BluRay.x264.DTS-FGT/After.the.Dark.2013.1080p.BluRay.x264.DTS-FGT.mkv.part": No file descriptors available (inout.c:101)
`

I'm also seeing related messages when I restart the container.
`
Caught SIGTERM signal!
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
s6-svwait: fatal: timed out
[s6-finish] sending all processes the TERM signal.
[2020-04-04 12:00:12.136] Reloading settings from "/config" (daemon.c:472)
[2020-04-04 12:00:12.136] Couldn't read "/config/settings.json": No file descriptors available (utils.c:258)
[2020-04-04 12:00:12.136] UDP Failed to set receive buffer: requested 4194304, got 3276800 (tr-udp.c:83)
[2020-04-04 12:00:12.136] UDP Please add the line "net.core.rmem_max = 4194304" to /etc/sysctl.conf (tr-udp.c:86)
`

I've tried setting the limits in my docker compose configuration.
`    ulimits:
      nproc: 65535
      nofile:
        soft: 50000
        hard: 100000
`

I can see during startup that the defined values are being overwritten within the container back from 50000 to 1024.
`
[2020-04-04 12:00:17.124] Transmission 2.94 (d8e60ee44f) started (session.c:740)
[2020-04-04 12:00:17.124] RPC Server Adding address to whitelist: 127.0.0.1 (rpc-server.c:971)
[2020-04-04 12:00:17.124] RPC Server Serving RPC and Web requests on port 127.0.0.1:9091/transmission/ (rpc-server.c:1213)
[2020-04-04 12:00:17.124] RPC Server Whitelist enabled (rpc-server.c:1217)
[2020-04-04 12:00:17.124] RPC Server Password required (rpc-server.c:1220)
[2020-04-04 12:00:17.124] UDP Failed to set receive buffer: requested 4194304, got 3276800 (tr-udp.c:83)
[2020-04-04 12:00:17.124] UDP Please add the line "net.core.rmem_max = 4194304" to /etc/sysctl.conf (tr-udp.c:86)
[2020-04-04 12:00:17.124] UDP Failed to set receive buffer: requested 4194304, got 3276800 (tr-udp.c:83)
[2020-04-04 12:00:17.124] UDP Please add the line "net.core.rmem_max = 4194304" to /etc/sysctl.conf (tr-udp.c:86)
[2020-04-04 12:00:17.124] DHT Reusing old id (tr-dht.c:307)
[2020-04-04 12:00:17.124] DHT Bootstrapping from 69 IPv4 nodes (tr-dht.c:156)
[2020-04-04 12:00:17.124] DHT Bootstrapping from 28 IPv6 nodes (tr-dht.c:159)
[2020-04-04 12:00:17.124] Using settings from "/config" (daemon.c:528)
[2020-04-04 12:00:17.124] Saved "/config/settings.json" (variant.c:1266)
[2020-04-04 12:00:17.124] transmission-daemon requiring authentication (daemon.c:554)
[2020-04-04 12:00:17.124] Watching "/watch" for new .torrent files (daemon.c:573)
[2020-04-04 12:00:17.124] Loaded 218 torrents (session.c:2034)
[2020-04-04 12:00:17.124] Port Forwarding (NAT-PMP) initnatpmp succeeded (0) (natpmp.c:70)
[2020-04-04 12:00:17.124] Port Forwarding (NAT-PMP) sendpublicaddressrequest succeeded (2) (natpmp.c:70)
[2020-04-04 12:00:18.124] Port Forwarding (UPnP) Found Internet Gateway Device "http://192.168.0.1:36507/ctl/IPConn" (upnp.c:225)
[2020-04-04 12:00:18.124] Port Forwarding (UPnP) Local Address is "192.168.0.5" (upnp.c:228)
[2020-04-04 12:00:18.124] Port Forwarding (UPnP) Port forwarding through "http://192.168.0.1:36507/ctl/IPConn", service "urn:schemas-upnp-org:service:WANIPConnection:1". (local address: 192.168.0.5:51413) (upnp.c:301)
[2020-04-04 12:00:18.124] Port Forwarding (UPnP) Port forwarding successful! (upnp.c:307)
[2020-04-04 12:00:18.124] Port Forwarding State changed from "Not forwarded" to "Forwarded" (port-forwarding.c:90)
[2020-04-04 12:00:18.124] Changed open file limit from 50000 to 1024 (fdlimit.c:380)
`

**Comment by drizuid** [comment]
@stevenboyd78 we do not allow showing piracy like this, please edit your post and remove the offending names.

**Comment by brosahay** [comment]
I am facing the same on raspberry pi 2, running Raspbian lite headless.

**Comment by TonyRL** [comment]
Check https://github.com/transmission/transmission/pull/893
It will be merged soon.

**Comment by 19wolf** [comment]
I've having this issue on 

Ubuntu 18.04
Docker version 19.03.11, build 42e35e61f3
Linux nephele 5.6.3-050603-generic #202004080837 SMP Wed Apr 8 08:39:44 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
[Docker Inspect](https://pastebin.com/WUDPQh4X)

**Comment by TimDowker** [comment]
Same issue here.  Has the merge been approved?

**Comment by gaozhitw** [comment]
I try to build own version (transmission 3.00+) to solve this problem.
https://github.com/gaozhitw/docker-transmission

**Comment by aptalca** [comment]
This image has been on 3+ for a while now

**Comment by maxcountryman** [comment]
Running into this issue on the latest tag.

**Comment by drizuid** [comment]
> Running into this issue on the latest tag: `transmission-daemon 3.00 (bb6b5a062e)`

please open a new issue and fill it out completely. you can reference this issue in your ticket but replying to something closed in 2020 is just silly.
